switchsdk/src/main/java/net/gini/switchsdk/TakePicturePresenter.java

```diff
@@ -117,6 +120,7 @@ public void onTakePictureSelected() {
             //if there are extractions available we finish the sdk
             if (mExtractionService.extractionsAvailable()) {
                 //TODO maybe add delay here
+                mDocumentService.cleanup();
```

Comment:
So much cleanup. Cleanup here, cleanup there, cleanup everywhere.
Can you combine it with the `editSdk`? It can be easy to forget to clean up in the future.